Index: data/static_data/pets/pet_merchant.xml
===================================================================
--- data/static_data/pets/pet_merchant.xml	(nonexistent)
+++ data/static_data/pets/pet_merchant.xml	(working copy)
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<merchants xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="pet_merchant.xsd">
+	<merchant id="1" rate_price="10"/>
+	<merchant id="2" rate_price="30"/>
+	<merchant id="3" rate_price="15"/>
+	<merchant id="4" rate_price="20"/>
+	<merchant id="5" rate_price="21"/>
+</merchants>
\ No newline at end of file
Index: data/static_data/pets/pet_merchant.xsd
===================================================================
--- data/static_data/pets/pet_merchant.xsd	(nonexistent)
+++ data/static_data/pets/pet_merchant.xsd	(working copy)
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<xs:schema version="1.0" xmlns:xs="http://www.w3.org/2001/XMLSchema">
+	<xs:element name="merchants">
+		<xs:complexType>
+			<xs:sequence>
+				<xs:element name="merchant" type="merch" minOccurs="0" maxOccurs="unbounded"/>
+			</xs:sequence>
+		</xs:complexType>
+	</xs:element>
+	<xs:complexType name="merch">
+		<xs:attribute name="id" type="xs:short"/>
+		<xs:attribute name="rate_price" type="xs:int"/>
+	</xs:complexType>
+</xs:schema>
Index: data/static_data/pets/pets.xml
===================================================================
--- data/static_data/pets/pets.xml	(revision 159)
+++ data/static_data/pets/pets.xml	(working copy)
@@ -951,7 +951,7 @@
         <petstats reaction="brave" run_speed="6.0" walk_speed="1.132" height="1.2"/>
     </pet>
     <pet id="900191" name="Royal Kitter" nameid="1613648" condition_reward="188051378">
-        <!-- <petfunction id="3" type="MERCHANT"/> -->
+        <petfunction id="3" type="MERCHANT"/>
         <petfunction id="1" type="LOOT"/>
         <petfunction id="1030000" type="WING"/>
         <petstats reaction="brave" run_speed="6.0" walk_speed="1.132" height="1.0"/>
@@ -963,7 +963,7 @@
         <petstats reaction="cowardly" run_speed="6.0" walk_speed="1.132" height="1.2"/>
     </pet>
     <pet id="900193" name="Kennercan" nameid="1613883" condition_reward="188051378">
-        <!-- <petfunction id="4" type="MERCHANT"/> -->
+        <petfunction id="4" type="MERCHANT"/>
         <petfunction id="1" type="LOOT"/>
         <petfunction id="1030000" type="WING"/>
         <petstats reaction="brave" run_speed="6.0" walk_speed="1.132" height="1.0"/>
@@ -1002,7 +1002,7 @@
         <petstats reaction="cowardly" run_speed="6.0" walk_speed="1.132" height="0.7"/>
     </pet>
     <pet id="900201" name="Mini Hyperion" nameid="1613977" condition_reward="188051378">
-        <!-- <petfunction id="4" type="MERCHANT"/> -->
+        <petfunction id="4" type="MERCHANT"/>
         <petfunction id="1" type="LOOT"/>
         <petfunction id="1030001" type="WING"/>
         <petstats reaction="brave" run_speed="6.0" walk_speed="1.132" height="0.7"/>
@@ -1050,7 +1050,7 @@
         <petstats reaction="brave" run_speed="6.0" walk_speed="1.132" height="1.0" altitude="1.5"/>
     </pet>
     <pet id="900210" name="Baby Hyperion" nameid="1614147" condition_reward="188051378">
-        <!-- <petfunction id="4" type="MERCHANT"/> -->
+        <petfunction id="4" type="MERCHANT"/>
         <petfunction id="1" type="LOOT"/>
         <petfunction id="1030001" type="WING"/>
         <petstats reaction="brave" run_speed="6.0" walk_speed="1.132" height="0.7"/>
@@ -1057,12 +1057,12 @@
     </pet>
     <pet id="970000" name="Sidekick 01" nameid="1609815" condition_reward="188051378">
         <petfunction id="1020013" type="BAG"/>
-        <!-- <petfunction id="1" type="MERCHANT"/> -->
+        <petfunction id="1" type="MERCHANT"/>
         <petfunction id="1030000" type="WING"/>
         <petstats reaction="brave" run_speed="6.0" walk_speed="1.132" height="1.0"/>
     </pet>
     <pet id="970001" name="Sidekick 02" nameid="1609816" condition_reward="188051378">
-        <!-- <petfunction id="2" type="MERCHANT"/> -->
+        <petfunction id="2" type="MERCHANT"/>
         <petstats reaction="brave" run_speed="6.0" walk_speed="1.132" height="1.0" altitude="1.5"/>
     </pet>
     <pet id="970002" name="Sidekick 03" nameid="1609817" condition_reward="188051378">
Index: data/static_data/pets/pets.xsd
===================================================================
--- data/static_data/pets/pets.xsd	(revision 159)
+++ data/static_data/pets/pets.xsd	(working copy)
@@ -40,7 +40,7 @@
             <xs:enumeration value="LOOT"/>
             <xs:enumeration value="DOPING"/>
 			<!--<xs:enumeration value="BUFF"/> Keine Funktion! -->
-			<!--<xs:enumeration value="MERCHANT"/> Keine Funktion! -->
+			<xs:enumeration value="MERCHANT"/>
         </xs:restriction>
     </xs:simpleType>
 </xs:schema>
\ No newline at end of file
Index: data/static_data/static_data.xml
===================================================================
--- data/static_data/static_data.xml	(revision 159)
+++ data/static_data/static_data.xml	(working copy)
@@ -91,6 +91,9 @@
     <dopings>
         <import file="pets/pet_doping.xml" skipRoot="true"/>
     </dopings>
+	<merchants>
+		<import file="pets/pet_merchant.xml" skipRoot="true"/>
+	</merchants>
     <import file="quest_data/quest_data.xml"/>
     <quest_scripts>
         <import file="quest_script_data" skipRoot="true"/>
Index: data/static_data/static_data.xsd
===================================================================
--- data/static_data/static_data.xsd	(revision 159)
+++ data/static_data/static_data.xsd	(working copy)
@@ -23,6 +23,7 @@
     <xs:include schemaLocation="pets/pets.xsd"/>
     <xs:include schemaLocation="pets/pet_feed.xsd"/>
     <xs:include schemaLocation="pets/pet_doping.xsd"/>
+	<xs:include schemaLocation="pets/pet_merchant.xsd"/>
     <xs:include schemaLocation="quest_data/quest_data.xsd"/>
     <xs:include schemaLocation="quest_script_data/quest_script_data.xsd"/>
     <xs:include schemaLocation="recipe/recipe_templates.xsd"/>
@@ -120,6 +121,7 @@
                 <xs:element ref="pets" minOccurs="0"/>
                 <xs:element ref="pet_feed" minOccurs="0"/>
                 <xs:element ref="dopings" minOccurs="0"/>
+				<xs:element ref="merchants" minOccurs="0"/>
                 <xs:element ref="quests" minOccurs="0"/>
                 <xs:element ref="quest_scripts" minOccurs="0"/>
                 <xs:element ref="recipe_templates" minOccurs="0"/>
Index: src/com/aionemu/gameserver/dataholders/DataManager.java
===================================================================
--- src/com/aionemu/gameserver/dataholders/DataManager.java	(revision 159)
+++ src/com/aionemu/gameserver/dataholders/DataManager.java	(working copy)
@@ -77,6 +77,7 @@
     public static PetData PET_DATA;
     public static PetFeedData PET_FEED_DATA;
     public static PetDopingData PET_DOPING_DATA;
+    public static PetMerchantData PET_MERCHANT_DATA;
     public static RoadData ROAD_DATA;
     public static InstanceCooltimeData INSTANCE_COOLTIME_DATA;
     public static DecomposableItemsData DECOMPOSABLE_ITEMS_DATA;
@@ -191,6 +192,7 @@
         PET_DATA = data.petData;
         PET_FEED_DATA = data.petFeedData;
         PET_DOPING_DATA = data.petDopingData;
+        PET_MERCHANT_DATA = data.petMerchantData;
         GUIDE_HTML_DATA = data.guideData;
         ROAD_DATA = data.roadData;
         INSTANCE_COOLTIME_DATA = data.instanceCooltimeData;
Index: src/com/aionemu/gameserver/dataholders/PetMerchantData.java
===================================================================
--- src/com/aionemu/gameserver/dataholders/PetMerchantData.java	(nonexistent)
+++ src/com/aionemu/gameserver/dataholders/PetMerchantData.java	(working copy)
@@ -0,0 +1,57 @@
+/**
+ * This file is part of Aion-Lightning <aion-lightning.org>.
+ *
+ *  Aion-Lightning is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Aion-Lightning is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details. *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Aion-Lightning.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.aionemu.gameserver.dataholders;
+
+import gnu.trove.map.hash.TShortObjectHashMap;
+
+import java.util.List;
+
+import javax.xml.bind.Unmarshaller;
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlRootElement;
+import javax.xml.bind.annotation.XmlTransient;
+
+import com.aionemu.gameserver.model.templates.pet.PetMerchantEntry;
+
+@XmlRootElement(name = "merchants")
+@XmlAccessorType(value=XmlAccessType.FIELD)
+public class PetMerchantData {
+
+    @XmlElement(name = "merchant")
+    private List<PetMerchantEntry> list;
+    @XmlTransient
+    private TShortObjectHashMap<PetMerchantEntry> merchantsById = new TShortObjectHashMap<PetMerchantEntry>();
+
+    void afterUnmarshal(Unmarshaller u, Object parent) {
+        for (PetMerchantEntry merch : list) {
+            merchantsById.put(merch.getId(), merch);
+        }
+
+        list.clear();
+        list = null;
+    }
+
+    public int size() {
+        return merchantsById.size();
+    }
+
+    public PetMerchantEntry getMerchantTemplate(short id) {
+        return merchantsById.get(id);
+    }
+}
Index: src/com/aionemu/gameserver/dataholders/StaticData.java
===================================================================
--- src/com/aionemu/gameserver/dataholders/StaticData.java	(revision 159)
+++ src/com/aionemu/gameserver/dataholders/StaticData.java	(working copy)
@@ -117,6 +117,8 @@
     public PetFeedData petFeedData;
     @XmlElement(name = "dopings")
     public PetDopingData petDopingData;
+    @XmlElement(name = "merchants")
+    public PetMerchantData petMerchantData;
     @XmlElement(name = "guides")
     public GuideHtmlData guideData;
     @XmlElement(name = "roads")
@@ -240,6 +242,7 @@
         DataManager.log.info("[DataManager] Loaded " + petData.size() + " pet templates");
         DataManager.log.info("[DataManager] Loaded " + petFeedData.size() + " food flavours");
         DataManager.log.info("[DataManager] Loaded " + petDopingData.size() + " pet doping templates");
+        DataManager.log.info("[DataManager] Loaded " + petMerchantData.size() + " pet merchant templates");
         DataManager.log.info("[DataManager] Loaded " + playerInitialData.size() + " initial player templates");
         DataManager.log.info("[DataManager] Loaded " + tradeListData.size() + " trade lists");
         DataManager.log.info("[DataManager] Loaded " + teleporterData.size() + " npc teleporter templates");
Index: src/com/aionemu/gameserver/model/templates/pet/PetMerchantEntry.java
===================================================================
--- src/com/aionemu/gameserver/model/templates/pet/PetMerchantEntry.java	(nonexistent)
+++ src/com/aionemu/gameserver/model/templates/pet/PetMerchantEntry.java	(working copy)
@@ -0,0 +1,40 @@
+/**
+ * This file is part of Aion-Lightning <aion-lightning.org>.
+ *
+ *  Aion-Lightning is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Aion-Lightning is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details. *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Aion-Lightning.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.aionemu.gameserver.model.templates.pet;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlAttribute;
+import javax.xml.bind.annotation.XmlType;
+
+@XmlType(name = "merch")
+@XmlAccessorType(value=XmlAccessType.NONE)
+public class PetMerchantEntry {
+
+    @XmlAttribute(name = "rate_price")
+    private int ratePrice;
+    @XmlAttribute(name = "id", required = true)
+    private short id;
+
+    public short getId() {
+        return id;
+    }
+
+    public int getRatePrice() {
+        return ratePrice;
+    }
+}
